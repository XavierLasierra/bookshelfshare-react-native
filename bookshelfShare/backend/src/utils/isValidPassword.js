const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
