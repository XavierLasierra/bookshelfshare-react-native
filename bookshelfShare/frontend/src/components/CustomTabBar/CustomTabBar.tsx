import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import BookIcon from '../../assets/bookIcon.svg';
import FollowingIcon from '../../assets/followingIcon.svg';
import ShelfIcon from '../../assets/shelfIcon.svg';
import ProfileIcon from '../../assets/profileIcon.svg';
import styles from './customTabBar.styles';
import AddPopUp from '../AddPopUp/AddPopUp';

export default function CustomTabBar({ state, navigation }: any) {
  function onPress(key: string, isFocused: boolean, name: string) {
    const event = navigation.emit({
      type: 'tabPress',
      target: key,
      canPreventDefault: true
    });

    if (!isFocused && !event.defaultPrevented) {
      navigation.navigate({
        name,
        merge: true
      });
    }
  }

  function onLongPress(key: string) {
    navigation.emit({
      type: 'tabLongPress',
      target: key
    });
  }

  return (
    <View style={styles.tabBarContainer}>
      <TouchableOpacity
        onPress={() => onPress(state.routes[0].key, state.index === 0, state.routes[0].name)}
        onLongPress={() => onLongPress(state.routes[0].key)}
        activeOpacity={0.8}
        style={[styles.tabOption, state.index === 0 && styles.tabOptionFocused]}
        testID="homeButton"
      >
        <BookIcon width={30} height={30} />
        <Text style={styles.tabText}>home</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => onPress(state.routes[1].key, state.index === 1, state.routes[1].name)}
        onLongPress={() => onLongPress(state.routes[1].key)}
        activeOpacity={0.8}
        style={[styles.tabOption, state.index === 1 && styles.tabOptionFocused]}
        testID="followingButton"
      >
        <FollowingIcon width={30} height={30} />
        <Text style={styles.tabText}>following</Text>
      </TouchableOpacity>
      <AddPopUp navigation={navigation} />
      <TouchableOpacity
        onPress={() => onPress(state.routes[2].key, state.index === 2, state.routes[2].name)}
        onLongPress={() => onLongPress(state.routes[2].key)}
        activeOpacity={0.8}
        style={[styles.tabOption, state.index === 2 && styles.tabOptionFocused]}
        testID="shelfButton"
      >
        <ShelfIcon width={30} height={30} />
        <Text style={styles.tabText}>shelf</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => onPress(state.routes[3].key, state.index === 3, state.routes[3].name)}
        onLongPress={() => onLongPress(state.routes[3].key)}
        activeOpacity={0.8}
        style={[styles.tabOption, state.index === 3 && styles.tabOptionFocused]}
        testID="profileButton"
      >
        <ProfileIcon width={30} height={30} />
        <Text style={styles.tabText}>profile</Text>
      </TouchableOpacity>
    </View>
  );
}
