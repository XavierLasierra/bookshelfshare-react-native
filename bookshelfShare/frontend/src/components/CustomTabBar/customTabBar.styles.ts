import {StyleSheet} from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  tabBarContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: stylesConstants.colors.white,
    height: 60,
  },
  tabOption: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 57,
    width: 57,
  },
  tabOptionFocused: {
    backgroundColor: stylesConstants.colors.light,
    borderRadius: 50,
  },
  tabText: {
    fontSize: 12,
    color: stylesConstants.colors.mainText,
  },
});
