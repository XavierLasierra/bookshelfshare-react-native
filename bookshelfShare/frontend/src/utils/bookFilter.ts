export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
