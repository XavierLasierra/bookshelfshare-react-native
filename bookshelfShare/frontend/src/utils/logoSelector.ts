import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
