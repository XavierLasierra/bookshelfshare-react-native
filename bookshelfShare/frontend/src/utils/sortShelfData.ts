export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
