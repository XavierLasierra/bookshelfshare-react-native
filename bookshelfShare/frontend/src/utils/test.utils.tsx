import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
