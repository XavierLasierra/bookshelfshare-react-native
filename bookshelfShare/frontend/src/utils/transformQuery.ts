interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
