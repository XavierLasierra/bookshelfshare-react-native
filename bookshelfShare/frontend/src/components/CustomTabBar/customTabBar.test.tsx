import React from 'react';
import { MenuProvider } from 'react-native-popup-menu';
import CustomTabBar from './CustomTabBar';
import { render, fireEvent } from '../../utils/test.utils';

const navigation = {
  navigate: jest.fn(),
  emit: jest.fn().mockReturnValue({ defaultPrevented: false })
};

describe('Given a CustomTabBar component', () => {
  describe('When it is rendered', () => {
    [0, 1, 2, 3].forEach((index) => {
      describe(`And state.index is ${index}`, () => {
        const state = {
          routes: [
            { key: 0 },
            { key: 1 },
            { key: 2 },
            { key: 3 }
          ],
          index
        };
        test('Then should match the snapshot', () => {
          const screen = render(
            <MenuProvider>
              <CustomTabBar state={state} navigation={navigation} />
            </MenuProvider>
          );
          expect(screen).toMatchSnapshot();
        });
      });
    });

    describe('And you press on homeButton which is currently focused', () => {
      const state = {
        routes: [
          { key: 0 },
          { key: 1 },
          { key: 2 },
          { key: 3 }
        ],
        index: 0
      };
      let screen;
      beforeEach(() => {
        screen = render(
          <MenuProvider>
            <CustomTabBar state={state} navigation={navigation} />
          </MenuProvider>
        );
        const button = screen.getByTestId('homeButton');
        fireEvent.press(button);
      });

      test('Then should call navigation.emit', () => {
        expect(navigation.emit).toHaveBeenCalled();
      });

      test('Then should not call navigation.navigate', () => {
        expect(navigation.navigate).not.toHaveBeenCalled();
      });
    });

    ['followingButton', 'shelfButton', 'profileButton'].forEach((buttonName) => {
      describe(`And you press on ${buttonName}`, () => {
        const state = {
          routes: [
            { key: 0 },
            { key: 1 },
            { key: 2 },
            { key: 3 }
          ],
          index: 0
        };
        let screen;
        beforeEach(() => {
          screen = render(
            <MenuProvider>
              <CustomTabBar state={state} navigation={navigation} />
            </MenuProvider>
          );
          const button = screen.getByTestId(buttonName);
          fireEvent.press(button);
        });

        test('Then should call navigation.emit', () => {
          expect(navigation.emit).toHaveBeenCalled();
        });

        test('Then should call navigation.navigate', () => {
          expect(navigation.navigate).toHaveBeenCalled();
        });
      });
    });

    ['homeButton', 'followingButton', 'shelfButton', 'profileButton'].forEach((buttonName) => {
      describe(`And you long press on ${buttonName}`, () => {
        const state = {
          routes: [
            { key: 0 },
            { key: 1 },
            { key: 2 },
            { key: 3 }
          ],
          index: 0
        };

        test('Then should call navigation.emit', () => {
          const screen = render(
            <MenuProvider>
              <CustomTabBar state={state} navigation={navigation} />
            </MenuProvider>
          );
          const button = screen.getByTestId(buttonName);
          fireEvent(button, 'longPress');
          expect(navigation.emit).toHaveBeenCalled();
        });
      });
    });
  });
});
