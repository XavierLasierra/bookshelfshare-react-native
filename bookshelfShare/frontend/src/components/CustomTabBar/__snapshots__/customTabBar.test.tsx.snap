// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CustomTabBar component When it is rendered And state.index is 0 Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 50,
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="homeButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          home
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="followingButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          following
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "elevation": 2,
            "height": 60,
            "justifyContent": "center",
            "position": "relative",
            "top": -10,
            "width": 60,
          }
        }
      >
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="openMenuButton"
          >
            <View
              ctx={
                Object {
                  "menuActions": Object {
                    "_getOpenedMenu": [Function],
                    "_notify": [Function],
                    "closeMenu": [Function],
                    "isMenuOpen": [Function],
                    "openMenu": [Function],
                    "toggleMenu": [Function],
                  },
                  "menuRegistry": Object {
                    "getAll": [Function],
                    "getMenu": [Function],
                    "setOptionsCustomStyles": [Function],
                    "subscribe": [Function],
                    "unsubscribe": [Function],
                    "updateLayoutInfo": [Function],
                  },
                }
              }
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3E3C37",
                    "fontSize": 30,
                    "height": 60,
                    "lineHeight": 60,
                    "textAlign": "center",
                    "width": 60,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="shelfButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          shelf
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="profileButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          profile
        </Text>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Given a CustomTabBar component When it is rendered And state.index is 1 Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="homeButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          home
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 50,
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="followingButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          following
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "elevation": 2,
            "height": 60,
            "justifyContent": "center",
            "position": "relative",
            "top": -10,
            "width": 60,
          }
        }
      >
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="openMenuButton"
          >
            <View
              ctx={
                Object {
                  "menuActions": Object {
                    "_getOpenedMenu": [Function],
                    "_notify": [Function],
                    "closeMenu": [Function],
                    "isMenuOpen": [Function],
                    "openMenu": [Function],
                    "toggleMenu": [Function],
                  },
                  "menuRegistry": Object {
                    "getAll": [Function],
                    "getMenu": [Function],
                    "setOptionsCustomStyles": [Function],
                    "subscribe": [Function],
                    "unsubscribe": [Function],
                    "updateLayoutInfo": [Function],
                  },
                }
              }
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3E3C37",
                    "fontSize": 30,
                    "height": 60,
                    "lineHeight": 60,
                    "textAlign": "center",
                    "width": 60,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="shelfButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          shelf
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="profileButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          profile
        </Text>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Given a CustomTabBar component When it is rendered And state.index is 2 Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="homeButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          home
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="followingButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          following
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "elevation": 2,
            "height": 60,
            "justifyContent": "center",
            "position": "relative",
            "top": -10,
            "width": 60,
          }
        }
      >
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="openMenuButton"
          >
            <View
              ctx={
                Object {
                  "menuActions": Object {
                    "_getOpenedMenu": [Function],
                    "_notify": [Function],
                    "closeMenu": [Function],
                    "isMenuOpen": [Function],
                    "openMenu": [Function],
                    "toggleMenu": [Function],
                  },
                  "menuRegistry": Object {
                    "getAll": [Function],
                    "getMenu": [Function],
                    "setOptionsCustomStyles": [Function],
                    "subscribe": [Function],
                    "unsubscribe": [Function],
                    "updateLayoutInfo": [Function],
                  },
                }
              }
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3E3C37",
                    "fontSize": 30,
                    "height": 60,
                    "lineHeight": 60,
                    "textAlign": "center",
                    "width": 60,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 50,
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="shelfButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          shelf
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="profileButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          profile
        </Text>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Given a CustomTabBar component When it is rendered And state.index is 3 Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="homeButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          home
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="followingButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          following
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "elevation": 2,
            "height": 60,
            "justifyContent": "center",
            "position": "relative",
            "top": -10,
            "width": 60,
          }
        }
      >
        <View
          collapsable={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="openMenuButton"
          >
            <View
              ctx={
                Object {
                  "menuActions": Object {
                    "_getOpenedMenu": [Function],
                    "_notify": [Function],
                    "closeMenu": [Function],
                    "isMenuOpen": [Function],
                    "openMenu": [Function],
                    "toggleMenu": [Function],
                  },
                  "menuRegistry": Object {
                    "getAll": [Function],
                    "getMenu": [Function],
                    "setOptionsCustomStyles": [Function],
                    "subscribe": [Function],
                    "unsubscribe": [Function],
                    "updateLayoutInfo": [Function],
                  },
                }
              }
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3E3C37",
                    "fontSize": 30,
                    "height": 60,
                    "lineHeight": 60,
                    "textAlign": "center",
                    "width": 60,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="shelfButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          shelf
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 50,
            "height": 57,
            "justifyContent": "center",
            "opacity": 1,
            "width": 57,
          }
        }
        testID="profileButton"
      >
        <IconMock
          height={30}
          width={30}
        />
        <Text
          style={
            Object {
              "color": "#3E3C37",
              "fontSize": 12,
            }
          }
        >
          profile
        </Text>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
